api/app/Http/Controllers/Forms/FormStatsController.php

```diff
@@ -18,9 +20,22 @@ public function getFormStats(Request $request)
         $form = $request->form; // Added by ProForm middleware
         $this->authorize('view', $form);
 
-        $formStats = $form->statistics()->where('date', '>', now()->subDays(29)->startOfDay())->get();
+        if (!$request->date_from || !$request->date_to) {
+            return $this->error([
+                'message' => 'Date range is required. Please select a date range.',
+            ]);
+        }
+
+        // Check if the date range is more than 3 months
+        if (Carbon::parse($request->date_from)->diffInMonths(Carbon::parse($request->date_to)) > 3) {
+            return $this->error([
+                'message' => 'Date range exceeds 3 months. Please select a shorter period.',
+            ]);
+        }
```

Comment:
It would be cleaner to create a dedicated request class for validation here (even if we use some custom validation in it)